target/linux/ramips/image/mt7621.mk

```diff
@@ -2562,14 +2538,15 @@ define Device/ubnt_edgerouter_common
   $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := Ubiquiti
-  IMAGE_SIZE := 256768k
+  IMAGE_SIZE := 259840k
   FILESYSTEMS := squashfs
-  KERNEL_SIZE := 3145728
-  KERNEL_INITRAMFS := $$(KERNEL) | \
-	ubnt-erx-factory-image $(KDIR)/tmp/$$(KERNEL_INITRAMFS_PREFIX)-factory.tar
```

Comment:
After removing specific image build target - size of image not checked at all.

remove KERNEL_SIZE define, set IMAGE_SIZE to 6144k, and modify ``IMAGE/sysupgrade.bin`` target to 
``IMAGE/sysupgrade.bin := sysupgrade-tar | check-size | append-metadata``

Reply:
Is the sysupgrade.bin limited to 6MB though?  I would have thought it isn't.

Reply:
kernel + squashfs image should fit in ``kernel`` partition. Or we need use mtdsplit partition scheme for floating ``ubi`` partition.

Reply:
factory flashing is not possible anymore as that is limited to 3MB still and no image can be built that small anymore. It is required to flash older version first or use tftp to install.

Reply:
Well, I'm wrong about _kernel + squashfs image should fit in kernel partition_, but simply set ``IMAGE_SIZE`` define - do nothing. At least - you need manually check via ``check-size`` size of root squashfs.

PS: And 259840+6144 is greater than usable RAM on device. and twice greater than /tmp partition.